### PR TITLE
Enhancement: bulk delete objects

### PR DESCRIPTION
**Pull request by shamoon**
## Proposed change

Sneak another one in. This feels like a thing we ought to have and works like one might expect. 

I renamed the API endpoint to `bulk_edit_objects` to make it more generic (now requires an `operation` parameter), so this is breaking for the API, hence api version increment.

Screenshot 2024-02-07 at 10 23 02 PM

Closes #2284
Closes #3029

## Type of change

- [ ] Bug fix: non-breaking change which fixes an issue.
- [x] New feature / Enhancement: non-breaking change which adds functionality. _Please read the important note above._
- [x] Breaking change: fix or feature that would cause existing functionality to not work as expected.
- [ ] Documentation only.
- [ ] Other. Please explain:

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Review by stumpylog — CHANGES_REQUESTED** [review]
Just a little tweak

File: src/documents/serialisers.py [hunk 1]
```diff
@@ -1352,11 +1362,13 @@ def _validate_permissions(self, permissions):
...
         object_type = attrs["object_type"]
         objects = attrs["objects"]
-        permissions = attrs["permissions"] if "permissions" in attrs else None
+        operation = attrs["operation"]
 
         self._validate_objects(objects, object_type)
-        if permissions is not None:
-            self._validate_permissions(permissions)
+        if operation == "set_permissions":
+            permissions = attrs["permissions"] if "permissions" in attrs else None
```
[the remark this answers is not in the document]
Comment (shamoon) [reply src/documents/serialisers.py]: Got it, you were right once I rebased. Should be sorted, thanks!

**Review by stumpylog — APPROVED** [review]

**Comment by stumpylog** [comment]
This might show up and two headings in the changelog, breaking and feature

**Comment by shamoon** [comment]
Yea, is that ok in this case (as opposed to when it puts things in fix + feature)?

**Comment by stumpylog** [comment]
Yeah, just kind of an FYI in case you wanted to adjust the labels

**Comment by hannibal85** [comment]
Hi @shamoon 
With current paperless-ngx 2.5.3 i have mis-behaviour of permissions.
Not really sure if this is really related to this PR as I never tried permissions with previous versions but I think so.

Situation:
I have a restricted user that only is allowed to show correspondents, tags, etc. but has no permissions to create, edit, delete
image

This user can not edit or delete single objects (as expected) but can use bulk delete and also change permissions via bulk edit.
image

**Comment by shamoon** [comment]
> Hi @shamoon With current paperless-ngx 2.5.3 i have mis-behaviour of permissions. Not really sure if this is really related to this PR as I never tried permissions with previous versions but I think so.

Yes, thanks. See https://github.com/paperless-ngx/paperless-ngx/pull/5888

**Comment by hannibal85** [comment]
Thanks for the quick fix. Awesome work.

**Pull request merged**